torchgeo/datasets/solar_plants_brazil.py

```diff
@@ -0,0 +1,176 @@
+    * Organized into `train`, `val`, and `test` splits
+    * Float32 GeoTIFF files for both input and mask images
+    * Spatial metadata included (CRS, bounding box), but not used directly for training
+
+    Folder structure:
+    root/train/input/img(123).tif
+    root/train/labels/target(123).tif
+
+    Access:
+    * Dataset is hosted on Hugging Face: https://huggingface.co/datasets/FederCO23/solar-plants-brazil
+    * Code and preprocessing steps available at: https://github.com/FederCO23/UCSD_MLBootcamp_Capstone
+
+    .. versionadded:: 0.8
+
+    Raises:
+        DatasetNotFoundError: If the dataset is not found and download=False.
+    """
+
+    url = 'https://huggingface.co/datasets/FederCO23/solar-plants-brazil/resolve/main/solarplantsbrazil.zip'
+    bands = ('Red', 'Green', 'Blue', 'NIR')
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+    ) -> None:
+        """Initialize a SolarPlantsBrazil dataset split.
+
+        Args:
+            root: Root directory where dataset is stored.
+            split: Dataset split to use, one of "train", "val", or "test".
+            transforms: Optional transforms to apply.
+            download: If True, download the dataset if it doesn't exist.
+        """
+        if split not in ['train', 'val', 'test']:
+            raise ValueError(
+                f"Invalid split '{split}', expected one of: 'train', 'val', or 'test'"
+            )
+
+        self.root = root
+        self.transforms = transforms
+        self.dataset_path = os.path.join(self.root, split)
+        self.split = split
+        self.download = download
+
+        self._verify()
+
+        self.image_paths = sorted(
+            glob.glob(os.path.join(self.dataset_path, 'input', 'img(*).tif'))
+        )
+        self.mask_paths = sorted(
+            glob.glob(os.path.join(self.dataset_path, 'labels', 'target(*).tif'))
+        )
+
+        if len(self.image_paths) == 0:
+            raise DatasetNotFoundError(self)
+
+        assert len(self.image_paths) == len(self.mask_paths), (
+            'Mismatch between image and mask files'
+        )
+
+    def _verify(self) -> None:
+        """Verify the dataset exists or download it."""
+        if os.path.exists(self.dataset_path) and os.listdir(self.dataset_path):
+            return
+
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        self._download()
+
+    def _download(self) -> None:
+        """Download the dataset archive from Hugging Face and extract it."""
+        download_and_extract_archive(
+            url=self.url, download_root=self.root, filename='solarplantsbrazil.zip'
+        )
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Retrieve an image-mask pair by index."""
```

Comment:
Every function docstring should define inputs (Args) and outputs (Returns) other than self/None